tests/test_trailedEdgeSources.py

```diff
@@ -231,8 +257,86 @@ def testEdgeFlag(self):
 
         # Test case with trailed source extending off chip.
         else:
+            self.assertEquals(record['truth_x'], -20)
             self.assertTrue(record.get("ext_trailedSources_Naive_flag_edge"))
             self.assertFalse(record.get("ext_trailedSources_Naive_flag"))
+            self.assertTrue(record.get("ext_trailedSources_Naive_flag_off_image"))
+            self.assertFalse(record.get("ext_trailedSources_Naive_flag_nan"))
+
+    def testNanFlag(self):
+        """Test if nan flags are correctly set in NaivePlugin.py
+
+        Given a `TrailedTestDataset`, run the NaivePlugin measurement which
+        has trailed sources where one of the end point values results in a
+        nan.
+        """
+        # Set up and run Naive measurement.
+        task = TrailedTaskSetup.makeTrailedSourceMeasurementTask(self,
+                                                                 plugin="ext_trailedSources_Naive",
+                                                                 dependencies=("base_SdssCentroid",
+                                                                               "base_SdssShape")
+                                                                 )
+
+        exposure, catalog = self.dataset.realize(5.0, task.schema, randomSeed=0)
+
+        original_check_edge_function = task.plugins['ext_trailedSources_Naive'].check_edge
+        # Used to simulate a trailed source where one of the coordinates is a
+        # nan.
+
+        def check_edge_mock(*args, **kwargs):
+            measRecord = args[0]
+            exposure = args[1]
+            x0 = args[2]
+            y0 = args[3]
+            x1 = args[4]
+            y1 = np.nan  # overriden to test NAN flagging
+            length = args[6]
+            measRecord['ext_trailedSources_Naive_y1'] = np.nan
+            return original_check_edge_function(measRecord, exposure, x0, y0, x1, y1, length)
+
+        patcher = patch(
+            'lsst.meas.extensions.trailedSources.NaivePlugin.SingleFrameNaiveTrailPlugin.check_edge',
```

Comment:
😂

Reply:
In all seriousness though, can you explain what you're mocking up here with a short comment, please?

Reply:
Added explanation for what the patcher is doing.